proposals/20210605-schema_curation_custom_component.md

```diff
@@ -0,0 +1,39 @@
+#### SIG TFX-Addons
+# Project Proposal
+
+**Your name:** Pratishtha Abrol
+
+**Your email:** pratishthaabrol@gmail.com
+
+**Your company/organization:** Outreachy
+
+**Project name:** [Schema curation custom component](https://github.com/tensorflow/tfx-addons/issues/8)
+
+## Project Description
+This project applies Python user code from a user-supplied module file to a schema produced by SchemaGen, to curate the schema based on domain knowledge.
+
+## Project Category
+Component
+
+## Project Use-Case(s)
+This project will allow the user to add a custom component that modifies the schema generated by SchemaGen component according to user knowledge, for example, fixing domain limits that were inferred wrongly by the SchemaGen component.
+
+## Project Implementation
+Implementation of the Schema Curation Custom Component can be done using the following approach:
+- Get the base Schema using SchemaGen component of TFX
```

Comment:
Could we include an example of fully generating the schema without the schemaGen programatically as well? 

If we make schema input channel optional, we can make this new component a generation component using user code.

Reply:
I'd suggest that we leave that as a follow-on project.  I don't think it will change the current implementation.